src/cobra/core/species.py

```diff
@@ -29,6 +29,10 @@ def __init__(self, id=None, name=None):
     def reactions(self):
         return frozenset(self._reaction)
 
+    @property
```

Comment:
Okay. I can remove this.
Species still has ._reaction, but there were various bits of code that were using ._reaction outside of Species, which I thought is frowned upon. This isn't actually necessary for GPR, I think.

Reply:
I think those are hidden attributes not private ones (the ones called `___name`). So they are okay to be used within the library but are not exposed outside of internal usage. Otherwise it become a pulic attribute and needs documentation etc.

Reply:
Okay. Corrected